### DISCUSSION TITLE
Trivy database not updated with fixed versions for libarchive-dev, libarchive13 & libcurl4

### DISCUSSION THREAD

**Discussion by venia-spai** [aquasecurity/trivy#5405] [Q&A] [answered]
### Description

Hello,

I am using trivy image scanning (currently only for critical vulnerabilities) and I have noticed that even after installing newer versions for libarchive-dev, libarchive13 & libcurl4, the scanning continues to be positive.
For example, in the aquasec page for CVE-2022-36227 (packages libarchive-dev and libarchive13) https://avd.aquasec.com/nvd/2022/cve-2022-36227/ it is mentioned that the vulnerability has been fixed since version 3.6.2.
Although the newly installed version of these packages is 3.7.2-1, the trivy scanning is still identifying the packages as vulnerable, as you can see in the attached image. The same applies to libcurl4, with installed version 8.4.0-2. In https://avd.aquasec.com/nvd/2023/cve-2023-23914/ it is mentioned that the vulnerability exists in curl <v7.88.0.

In the report you can see that the "Fixed version" field is blank, so I assume the DB has not been updated to include the package versions which are not affected by the vulnerability.

![image](https://github.com/aquasecurity/trivy/assets/84719106/15d5e086-6df1-44ae-a0d9-eb6384c7a0b3)

Also, is there maybe a temporary workaround (e.g. configure trivyignore to ignore specific package with specific version)? I couldn't find any such option yet, but it would be great to add a feature like this

### Desired Behavior

After updating the packages, the image vulnerability scanning should not report any vulnerabilities.

### Actual Behavior

The image vulnerability scanning is still reporting vulnerabilities.
![image](https://github.com/aquasecurity/trivy/assets/84719106/373d5621-4f13-4f8b-8e93-a385c8ff8aa9)

### Reproduction Steps

```bash
1. Updated the packages to versions which are not affected according to the avd.aquasec.com page for each CVE
2. Reran the "trivy image --scanners vuln --severity CRITICAL <image:tag>"  command
3. The trivy report is still identifying the same vulnerabilities
```

### Target

Container Image

### Scanner

Vulnerability

### Output Format

None

### Mode

Standalone

### Debug Output

```bash
Output of command "trivy image --scanners vuln --severity CRITICAL --debug <image:tag>"


2023-10-18T15:30:37.408Z	DEBUG	Missing diff ID in cache: sha256:cd5d0a5ef3063d21faad26f1b0d0c1b91ef07353e50ee432d77b16e77a85f7bc
2023-10-18T15:30:37.554Z	DEBUG	Unable to parse "var/lib/dpkg/available" file: file open error: open var/lib/dpkg/available: file does not exist
2023-10-18T15:30:37.564Z	DEBUG	Missing diff ID in cache: sha256:2c1f011db826f78035512c3f3e8d6cec8416994f8f2ffaa7a7f3c03c1f274498
2023-10-18T15:30:37.666Z	DEBUG	Unable to parse "var/lib/dpkg/available" file: file open error: open var/lib/dpkg/available: file does not exist
2023-10-18T15:30:37.681Z	DEBUG	Missing diff ID in cache: sha256:e0d1289d0d95a74d34f134199a51418880093e76f9bd9b488e8181f634bbb237
2023-10-18T15:30:37.821Z	DEBUG	Missing diff ID in cache: sha256:d665294ed04aca3cea0f2d994d738b00c42862711eb40d403844c4a1219c1b36
2023-10-18T15:30:37.920Z	DEBUG	Missing diff ID in cache: sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef
2023-10-18T15:30:37.943Z	DEBUG	Missing diff ID in cache: sha256:d0d33e707617846d897c4a0efd697ef1f84242e872436f3574b6bab48e13cbc1
2023-10-18T15:30:39.947Z	DEBUG	Unable to parse "var/lib/dpkg/available" file: file open error: open var/lib/dpkg/available: file does not exist
2023-10-18T15:30:41.848Z	INFO	Detected OS: debian
2023-10-18T15:30:41.849Z	INFO	Detecting Debian vulnerabilities...
2023-10-18T15:30:41.849Z	DEBUG	debian: os version: 11
2023-10-18T15:30:41.849Z	DEBUG	debian: the number of packages: 210
2023-10-18T15:30:41.905Z	INFO	Number of language-specific files: 3
2023-10-18T15:30:41.905Z	INFO	Detecting dotnet-core vulnerabilities...
2023-10-18T15:30:41.905Z	DEBUG	Detecting library vulnerabilities, type: dotnet-core, path: usr/share/dotnet/shared/Microsoft.NETCore.App/6.0.23/Microsoft.NETCore.App.deps.json
2023-10-18T15:30:41.905Z	DEBUG	Detecting library vulnerabilities, type: dotnet-core, path: usr/share/dotnet/shared/Microsoft.AspNetCore.App/6.0.23/Microsoft.AspNetCore.App.deps.json
2023-10-18T15:30:41.907Z	DEBUG	Detecting library vulnerabilities, type: dotnet-core, path: app/Nbg.NetCore.DocOCRInfoExtraction.deps.json
```

### Operating System

Ubuntu 22.04.3 LTS

### Version

```bash
v0.46.0
Also tested in v.0.44.0
```

### Checklist

- [ ] Run `trivy image --reset`
- [ ] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by DmitriyLewen** [comment]
Hello @venia-spai 
Thanks for your report!

We use [Ubuntu CVE tracker](https://ubuntu.com/security/cves) for Ubuntu images. More information about Data Source Selection here - https://aquasecurity.github.io/trivy/v0.46/docs/scanner/vulnerability/#data-source-selection

Looks like Ubuntu doesn't have fix for [CVE-2022-36227](https://ubuntu.com/security/CVE-2022-36227).

FYI - Ubuntu CVE tracker uses source packages (in your case source package for `libarchive-dev` and `libarchive13` is `libarchive` - https://packages.ubuntu.com/focal/libarchive13).
You can see names of source packages with `-f json --list-all-pkgs` flags in `Packages` array.

Regards, Dmitriy

**Reply by venia-spai** [reply]
Hello @DmitriyLewen,

Thanks for your prompt response! Actually, the OS of the image that I am scanning is Debian 11. I mentioned Ubuntu here in the discussion because I thought that you needed the OS of the machine that the Trivy is running on to scan the image.
I assume you are following the same process for Debian packages? (i.e. https://security-tracker.debian.org )
If this is the case I can see here https://security-tracker.debian.org/tracker/CVE-2022-36227 that the CVE is fixed in the version that I have installed.

Best Regards,
Venia

**Reply by DmitriyLewen** [reply]
> If this is the case I can see here https://security-tracker.debian.org/tracker/CVE-2022-36227 that the CVE is fixed in the version that I have installed.

But `CVE-2022-36227` for [Debian bullseye](https://www.debian.org/News/2021/20210814) is marked as vulnerable:
![изображение](https://github.com/aquasecurity/trivy/assets/91113035/bee2d545-09b4-43a7-b08c-4b031db68fe2)

**Reply by venia-spai** [reply]
I can see in the debian page  https://security-tracker.debian.org/tracker/CVE-2022-36227 that the bookworm distribution has fixed this with version 3.6.2-1. Bookworm is the current stable distribution, shouldn't this be reflected on Trivy library as fixed version for this CVE?
Perhaps Trivy is using only bullseye as a reference?

**Reply by DmitriyLewen** [reply]
We use separate advisory list for each Debian releases(just like Debian itself). 
Because for each release, vendors create separate patches.

In this case Debian fixed this vulnerability in `3.6.2-1` for `bookworm` only.
But it does not mean that `3.6.2-1` for `bullseye` or other releases doesn't contain `CVE-2022-36227`.

**Reply by venia-spai** [reply]
Thanks a lot for the information. So, if I install libarchive version 3.6.2-1 from bookworm release, the expected behavior is that this vulnerability will not show up in the Trivy report, is my understanding correct?

**Reply by DmitriyLewen** [reply]
Unfortunately, we can't determine which version this package was created for.

We use the following logic:
Trivy finds your OS (Debian-11 in your case), takes advisory list for that OS, and compares the package versions it finds and the versions on the advisory list.

> if I install libarchive version 3.6.2-1 from bookworm release, the expected behavior is that this vulnerability will not show up in the Trivy report, is my understanding correct?

Trivy still shows this vulnerability

**Reply by venia-spai** [reply]
OK, I understand, is there perhaps a workaround solution for us, like using trivyignore to exclude a specific version for a specific package, in our case this means ignoring package libarchive version 3.6.2-1 ?
This of cource would imply that we have ensured internally that this version has been installed from bookworm release.

**Reply by DmitriyLewen** [reply, marked as answer]
You can ignore `CVE-2022-36227` with `.trivyignore` file, but you can't associate this CVE with package version.

For your case you can create [module](https://aquasecurity.github.io/trivy/v0.46/docs/advanced/modules/).
I think you can use the following logic for the module:
Module will find `CVE-2022-36227` in result and check package version. If version is `3.6.2-1` - modules will remove this vulnerability from result.

**Reply by venia-spai** [reply]
OK, thank you very much for your help! So, I am closing this discussion
